File: src/main/java/com/uwetrottmann/trakt/v2/entities/ShowIds.java
package com.uwetrottmann.trakt.v2.entities;

public class ShowIds extends BaseIds {

    public String slug;
    public Integer tvdb;
    public Integer tvrage;

    public static ShowIds trakt(int trakt) {
        ShowIds ids = new ShowIds();
        ids.trakt = trakt;
        return ids;
    }

    public static ShowIds imdb(String imdb) {
        ShowIds ids = new ShowIds();
        ids.imdb = imdb;
        return ids;
    }

    public static ShowIds tmdb(int tmdb) {
        ShowIds ids = new ShowIds();
        ids.tmdb = tmdb;
        return ids;
    }

    public static ShowIds slug(String slug) {
        ShowIds ids = new ShowIds();
        ids.slug = slug;
        return ids;
    }

    public static ShowIds tvdb(int tvdb) {
        ShowIds ids = new ShowIds();
        ids.tvdb = tvdb;
        return ids;
    }

    public static ShowIds tvrage(int tvrage) {
        ShowIds ids = new ShowIds();
        ids.tvrage = tvrage;
        return ids;
    }

}

File: src/main/kotlin/com/redissi/trakt/entities/ShowIds.kt
package com.redissi.trakt.entities

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
class ShowIds(
    override val trakt: Int? = null,
    override val imdb: String? = null,
    override val tmdb: Int? = null,
    val slug: String? = null,
    val tvdb: Int? = null,
    val tvrage: Int? = null
) : BaseIds() {

    companion object {
        fun trakt(trakt: Int): ShowIds {
            return ShowIds(trakt = trakt)
        }

        fun imdb(imdb: String?): ShowIds {
            return ShowIds(imdb = imdb)
        }

        fun tmdb(tmdb: Int): ShowIds {
            return ShowIds(tmdb = tmdb)
        }

        fun slug(slug: String?): ShowIds {
            return ShowIds(slug = slug)
        }

        fun tvdb(tvdb: Int): ShowIds {
            return ShowIds(tvdb = tvdb)
        }

        fun tvrage(tvrage: Int): ShowIds {
            return ShowIds(tvrage = tvrage)
        }
    }
}